File: project/app/src/main/java/com/marcosvbras/empresas/views/utils/ErrorObservable.java
package com.marcosvbras.empresas.views.utils;

import android.databinding.BaseObservable;

public class ErrorObservable extends BaseObservable {

    private String error;
    private Integer errorString;

    public ErrorObservable() {}

    public ErrorObservable(String error) {
        this.error = error;
    }

    public String get() {
        return error;
    }

    public void set(String error) {
        this.error = error;
        notifyChange();
    }

    public Integer getIntError() {
        return errorString;
    }

    public void set(Integer errorString) {
        this.errorString = errorString;
        notifyChange();
    }

    public void clear() {
        this.error = null;
        this.errorString = null;
        notifyChange();
    }

    public boolean hasError() {
        return error != null || errorString != null;
    }
}

File: project/app/src/main/java/com/marcosvbras/empresas/utils/ErrorObservable.kt
package com.marcosvbras.empresas.utils

import android.databinding.BaseObservable

class ErrorObservable() : BaseObservable() {

    constructor(error: String) : this() {
        this.error = error
        notifyChange()
    }

    constructor(error: Int) : this() {
        this.errorInt = error
        notifyChange()
    }

    var error: String? = null
        set(value) {
            field = value
            notifyChange()
        }

    var errorInt: Int? = null
        set(value) {
            field = value
            notifyChange()
        }

    fun clear() {
        error = null
        errorInt = null
        notifyChange()
    }

    fun hasError() : Boolean = error != null || errorInt != null
}